wasm/src/lib.rs

```diff
@@ -85,40 +87,33 @@ impl ExecuteOnHost for data_model::query::QueryBox {
     }
 }
 
-/// Calculate the result of the expression on the host side without mutating the state.
-pub trait EvaluateOnHost {
-    /// The resulting type of the expression.
-    type Value;
-    /// Type of error
-    type Error;
-
-    /// Calculate the result on the host side.
-    ///
-    /// # Errors
-    ///
-    /// Depends on the implementation.
-    fn evaluate(&self) -> Result<Self::Value, Self::Error>;
+/// Context of expression evaluation
+#[derive(Clone, Default)]
+#[repr(transparent)]
+pub struct Context {
+    values: BTreeMap<Name, Value>,
 }
 
-impl<V: TryFrom<Value> + DecodeAll> EvaluateOnHost for EvaluatesTo<V> {
-    type Value = V;
-    type Error = <V as TryFrom<Value>>::Error;
+impl Context {
+    /// Create new [`Self`]
+    pub fn new() -> Self {
+        Self {
+            values: BTreeMap::new(),
+        }
+    }
+}
 
-    fn evaluate(&self) -> Result<Self::Value, Self::Error> {
-        #[cfg(not(test))]
-        use host::evaluate_on_host as host_evaluate_on_host;
-        #[cfg(test)]
-        use tests::_iroha_wasm_evaluate_on_host_mock as host_evaluate_on_host;
+impl iroha_data_model::evaluate::Context for Context {
+    fn query(&self, query: &QueryBox) -> Result<Value, QueryExecutionFailure> {
+        Ok(query.execute())
+    }
```

Comment:
I can't see a reason to not to allow user to just call `query.execute()` in its own. So why do we need this?

Reply:
look [here](https://github.com/hyperledger/iroha/pull/3351/files#r1157745952)

Reply:
Okay, makes sense